vmdb/app/controllers/container_controller.rb

```diff
@@ -0,0 +1,360 @@
+      self.x_active_accord ||= 'containers'
+      @built_trees.push(build_containers_tree)
+      @accords.push(:name      => "containers",
+                    :title     => "Relationships",
+                    :container => "containers_tree_div")
+    end
+
+    if role_allows(:feature => "container_filter_accord", :any => true)
+      self.x_active_tree   ||= 'containers_filter_tree'
+      self.x_active_accord ||= 'containers_filter'
+      @built_trees.push(build_containers_filter_tree)
+      @accords.push(:name      => "containers_filter",
+                    :title     => "All Containers",
+                    :container => "containers_filter_tree_div")
+    end
+
+    params.merge!(session[:exp_parms]) if session[:exp_parms]  # Grab any explorer parm overrides
+    session.delete(:exp_parms)
+
+    get_node_info(x_node)
+    @in_a_form = false
+
+    render :layout => "explorer"
+  end
+
+  def identify_container(id = nil)
+    @container = @record = identify_record(id || params[:id])
+  end
+
+  # ST clicked on in the explorer right cell
+  def x_show
+    identify_container(from_cid(params[:id]))
+    respond_to do |format|
+      format.js do                  # AJAX, select the node
+        @explorer = true
+        params[:id] = x_build_node_id(@record,nil,x_tree(:containers_tree))  # Get the tree node id
+        tree_select
+      end
+      format.html do                # HTML, redirect to explorer
+        tree_node_id = TreeBuilder.build_node_id(@record)
+        session[:exp_parms] = {:id => tree_node_id}
+        redirect_to :action => "explorer"
+      end
+      format.any {render :nothing => true, :status => 404}
+    end
+  end
+
+  # Tree node selected in explorer
+  def tree_select
+    @explorer = true
+    @lastaction = "explorer"
+    self.x_node = params[:id]
+    @nodetype, id = params[:id].split("_").last.split("-")
+
+    if x_tree[:type] == :containers_filter && TreeBuilder.get_model_for_prefix(@nodetype) != "Container"
+      search_id = @nodetype == "root" ? 0 : from_cid(id)
+      adv_search_build(vm_model_from_active_tree(x_active_tree))
+      session[:edit] = @edit              # Set because next method will restore @edit from session
+      listnav_search_selected(search_id) unless params.has_key?(:search_text) # Clear or set the adv search filter
+      if @edit[:adv_search_applied] &&
+        MiqExpression.quick_search?(@edit[:adv_search_applied][:exp]) &&
+        %w(reload tree_select).include?(params[:action])
+        self.x_node = params[:id]
+        quick_search_show
+        return
+      end
+    end
+
+    replace_right_cell
+  end
+
+  # Accordion selected in explorer
+  def accordion_select
+    @layout     = "explorer"
+    @lastaction = "explorer"
+    self.x_active_accord = params[:id]
+    self.x_active_tree   = "#{params[:id]}_tree"
+    replace_right_cell
+  end
+
+  private
+
+  def get_record_display_name(record)
+    record.name
+  end
```

Comment:
@tenderlove, some of these methods are copied over from other explorer but will be used when @erinboyd starts enhancing these screens which is why i left them in there.